### PR TITLE
Accommodate separate lmod, tcl modules.yaml's

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
## Description

This PR updates the 'create env' command in order to support the separation of common/modules.yaml into lmod and tcl ones.

## Issue(s) addressed

Addresses https://github.com/JCSDA/spack-stack/issues/703, https://github.com/JCSDA/spack-stack/issues/50

## Dependencies

none

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have run the unit tests before creating the PR

### CODE REVIEW

**Review by srherbener — APPROVED** [review]
These changes look okay to me, but I haven't tested. I'm relying on the CI testing.

File: lib/jcsda-emc/spack-stack/stack/stack_env.py [hunk 1]
```diff
@@ -108,11 +109,39 @@ def env_dir(self):
...
+        site_modules_yaml_path = os.path.join(site_configs_dir, "modules.yaml")
+        with open(site_modules_yaml_path, "r") as f:
+            site_modules_yaml = syaml.load_config(f)
+        lmod_or_tcl_list = site_modules_yaml["modules"]["default"]["enable"]
+        assert (
+            lmod_or_tcl_list
+        ), "Set 'modules:default:enable' in site modules.yaml, or use --modulesys {tcl,lmod}"
+        if len(set(lmod_or_tcl_list)) > 1:
+            logging.warning(
+                "WARNING: Multiple lmod/tcl settings found for %s; using the first"
```
Comment (climbfuji) [inline lib/jcsda-emc/spack-stack/stack/stack_env.py]: I think this should be an error that leads to an abort.

**Review by climbfuji — COMMENTED** [review]
This is great, thanks. Any possibilities to include it in the unit tests here and also in the spack-stack unit tests?

File: lib/jcsda-emc/spack-stack/stack/stack_env.py (hunk 1 above)
Comment (AlexanderRichert-NOAA) [reply lib/jcsda-emc/spack-stack/stack/stack_env.py]: Okay sounds good

**Comment by AlexanderRichert-NOAA** [comment]
@climbfuji I don't see the jcsda-emc unit tests getting run anywhere, am I missing something? I added a line to unit_tests.yaml just now that I think will run them. EDIT: nevermind, got it :)

**Comment by climbfuji** [comment]
This looks great, I am testing this now on my macOS.

**Review by climbfuji — CHANGES_REQUESTED** [review]
There's a problem with spack parsing the double colons in the site modules.yaml file, see attached screenshot (from my macOS where the default is lmod, i.e. an unchanged environment (no `--modulesys` option) is fine, but the modified env is not.

Screen Shot 2023-10-09 at 8 41 28 AM

**Comment by AlexanderRichert-NOAA** [comment]
I saw the same thing in my testing, but it didn't seem to negatively impact anything. Is it breaking the test environment/keeping you from installing?

**Comment by climbfuji** [comment]
> I saw the same thing in my testing, but it didn't seem to negatively impact anything. Is it breaking the test environment/keeping you from installing?

I haven't tried that, but I know we added some logic for the containers previously (I might have been able to remove that in the last container PR, don't remember) that replaced the double colons. I can give that a try, but it would be cleaner if we didn't have these malformatted yaml files.

**Comment by AlexanderRichert-NOAA** [comment]
Okay. I'd prefer to leave as-is if we can, but if needed we can always run a search and replace on the output file to get rid of the apostrophes.

**Comment by climbfuji** [comment]
I found it, it’s still there. Check https://github.com/JCSDA/spack/blob/8296fcf898cb2997ad309273ab8480114c134bca/lib/jcsda-emc/spack-stack/stack/container_env.py#L52-L57 <https://github.com/JCSDA/spack/blob/8296fcf898cb2997ad309273ab8480114c134bca/lib/jcsda-emc/spack-stack/stack/container_env.py#L52-L57> it’s actually quite simple. Any chance you can add that?

**Review by climbfuji — APPROVED** [review]
Perfect, thanks very much for this. Works like a charm on my macOS laptop.

**Pull request merged**